theano/tensor/basic.py

```diff
@@ -4078,6 +4078,79 @@ def R_op(self, inputs, eval_points):
             return [None]
         return self.make_node(*eval_points).outputs
 
+    def c_code_cache_version(self):
+        return (1,)
+
+    def c_code(self, node, name, inputs, outputs, sub):
+        x, = inputs
+        out, = outputs
+        outdim = self.outdim
+        fail = sub['fail']
+        return """
+        if (%(outdim)s == PyArray_NDIM(%(x)s))
+        {
+            if (NULL != %(out)s && %(x)s != %(out)s)
+                Py_XDECREF(%(out)s);
+
+            %(out)s = %(x)s;
```

Comment:
Ah, I see, you would completely omit checking if `%(out)s` and `%(x)s` are the same already.